tests/testtortransport.nim

```diff
@@ -162,3 +167,125 @@ suite "Tor transport":
     "/ip4/127.0.0.1/tcp/8080/onion3/a2mncbqsbullu7thgm4e6zxda2xccmcgzmaq44oayhdtm6rav5vovcad:80",
     "/ip4/127.0.0.1/tcp/8081/onion3/a2mncbqsbullu7thgm4e6zxda2xccmcgzmaq44oayhdtm6rav5vovcae:81",
   )
+
+proc testTransportEvents(transport: Transport, addrs: seq[MultiAddress]) {.async.} =
+  ## Common test procedure for transport events
+  var onRunningFired = false
+  var onStopFired = false
+
+  proc onRunningHandler() {.async.} =
+    # onRunning only gets set during start() call
+    if transport.onRunning.isNil:
+      transport.onRunning = newAsyncEvent()
+    await transport.onRunning.wait()
+    onRunningFired = true
+
+  proc onStopHandler() {.async.} =
+    # onStop only gets set during stop() call
+    if transport.onStop.isNil:
+      transport.onStop = newAsyncEvent()
+    await transport.onStop.wait()
+    onStopFired = true
+
+  asyncSpawn onRunningHandler()
+  asyncSpawn onStopHandler()
+
+  check:
+    onRunningFired == false
+    onStopFired == false
+    transport.running == false
+
+  await transport.start(addrs)
+
+  # Give the event handler time to run
+  await sleepAsync(10.milliseconds)
+
+  check:
+    onRunningFired == true
+    onStopFired == false
+    transport.running == true
+
+  await transport.stop()
+
+  # Give the event handler time to run  
+  await sleepAsync(10.milliseconds)
+
+  check:
+    onRunningFired == true
+    onStopFired == true
+    transport.running == false
+
+suite "Transport Events":
+  asyncTest "onRunning with TCP transport":
+    let upgrader = Upgrade()
+    let transport = TcpTransport.new(upgrade = upgrader)
```

Comment:
nit: inline upgrader since variable is not needed and it line length of next line almost the same.
 `let transport = TcpTransport.new(upgrade = Upgrade())`
 
 do this in other tests as well.